tests/browser/test_observe.py

```
@@ -0,0 +1,288 @@
+import asyncio
+import datetime as dt
+import json
+import re
+from collections.abc import Sequence
+from dataclasses import asdict
+from pathlib import Path
+from typing import Any, Final
+from urllib.parse import urlparse
+
+from notte_core.utils import url
```

Ruff (0.12.2)

`notte_core.utils.url` imported but unused

Remove unused import: `notte_core.utils.url`

(F401)